### ISSUE TITLE
m4a artist ID is is mistakenly output as AlbumTitleID

### ISSUE THREAD

**Issue by malhal** [exiftool/exiftool#216]
Hi while working with a `.m4a` from Apple Music I noticed a mistake in one of the tags output. `AlbumTitleID` is mistakenly showing the iTunes Artist ID, e.g.
```
exiftool she-taught-me-how-to-fly.m4a -AlbumTitleID
Album Title ID                  : 450757148
```
At that number output is an artist ID, as you can verify at the following link:
https://music.apple.com/us/artist/450757148

I believe I tracked the problem down to this line of QuickTime.pm:
https://github.com/exiftool/exiftool/blob/d35e9e26e0a8b443dae307f55d0a4a067d311a16/lib/Image/ExifTool/QuickTime.pm#L3399C1-L3403C7
```
    atID => { #10 (or TV series)
        Name => 'AlbumTitleID',
        Format => 'int32u',
        Writable => 'int32s', #27
    },
```
The mistake is in the Name, 'AlbumTitleID' should be `ArtistID` (or `ArtistTitleID` if you prefer). May need an `if MP4`?

Ref
https://docs.mp3tag.de/mapping/
Tag fields exclusive to MP4
`ITUNESARTISTID | atID | iTunes Artist ID`

Also, the plID (Play List ID) which is also tag for the iTunes Album ID, is being output as an array of bytes intead of the integer, i.e. I'm seeing
```
Play List ID                    : 0 0 0 0 90 117 21 79
```
where as I expect to see
```
Play List ID                    : 1517622607
```
From the spec:
```
ITUNESALBUMID	plID	iTunes Album ID
```

Just to clarify, the file I'm testing has the following MP4 tags:
```
Atom "atID" contains: 450757148 (iTunes Artist ID)
Atom "plID" contains: 1517622607 (iTunes Album ID)
```

**Comment by boardhead** [comment]
Thanks.  I'll look into this.

- Phil

**Comment by boardhead** [comment]
Looks good.  I'll change these to ArtistID and AlbumID.  I couldn't download your sample... Could you post the exiftool -v3 output for the PlayListID (soon to be AlbumID), I want to verify the format for writing this tag.

Also, I'll add ComposerID (cmID), but I don't have a sample available so I'll need the -v3 output for this tag to be able to add the write ability.  Do you have such a sample?

- Phil

**Comment by boardhead** [comment]
Implemented in version 12.65 on Aug. 20, 2023
